vllm/distributed/parallel_state.py

```diff
@@ -674,6 +678,30 @@ def model_parallel_is_initialized():
     return (_TP is not None and _PP is not None)
 
 
+OVERRIDE_TP_STATE = False
+
+
+@contextlib.contextmanager
+def patch_tensor_parallel_group(world_group, tp_group):
+    """Patch the tp group temporarily until this function ends."""
+    global OVERRIDE_TP_STATE
+    if not OVERRIDE_TP_STATE and world_group and tp_group:
+        OVERRIDE_TP_STATE = True
+        old_world_group = get_world_group()
```

Comment:
Yes, it is used only during initialization.
And if we do not override world_group then it will fail during initialization due to an assertion check ([link](https://github.com/vllm-project/vllm/pull/5414/files#diff-cf486d32b11454e89103000ac0f9a93ee1ccbb8983c02e794b734a65301cd98bR566)).
This checks the world_group size consistency between workers when spawning multiple workers in the same process (or ray worker).
In our case, it asserts the existing world_group (target's) should have the same size with the world_group being initialized (draft's).
Note that this check is added by #5293 and slightly modified in this PR to support the small draft-tp case.
I'm not sure in what situations this check is used, but I thought it would be safer to keep it.

It's a slightly different story, but when I opened this PR, things were a little different.
get_world_group() was also used after initialization by broadcast_tensor_dict(), which is for driver to control workers.
But this use case has gone after refactoring by the other PR.

Reply:
Thanks for the explanation.

How about add a comment about world_group? Since this function is named as `patch_tensor_parallel_group`. Or maybe rename the function to `patch_distributed_group`?

Reply:
Good suggestion :)
I've added a comment in the docstring of `patch_tensor_parallel_group()`.